mono/metadata/cominterop.c

```diff
@@ -573,22 +626,16 @@ cominterop_get_interface_checked (MonoComObject* obj, MonoClass* ic, MonoError *
  * Returns: the COM interface requested
  */
 static gpointer
-cominterop_get_interface (MonoComObject *obj, MonoClass *ic, gboolean throw_exception)
+cominterop_get_interface (MonoComObject *obj, MonoClass *ic)
 {
 	ERROR_DECL (error);
 	gpointer itf = cominterop_get_interface_checked (obj, ic, error);
 	if (!is_ok (error)) {
-		if (throw_exception) {
-			mono_error_set_pending_exception (error);
-			return NULL;
-		} else {
-			mono_error_cleanup (error);
-		}
-	}
-
-	if (throw_exception)
+		mono_error_set_pending_exception (error);
+		itf = NULL;
```

Comment:
This is incorrect or at least unclear to me now.
It should use goto exit, and then leave alone the if (throw_exception) g_assert(itf).
Point being -- single point of exit, because we will have function enter/exit, but don't change more than that.